include/odp/api/spec/schedule.h

```diff
@@ -45,6 +46,53 @@ extern "C" {
  * Maximum schedule group name length in chars including null char
  */
 
+/**
+ * Schedule configuration options
+ */
+typedef struct odp_schedule_config_t {
+
+	/* Scheduler flow aware
+	 * Application can configure the scheduler to be flow aware or unaware.
+	 * When the scheduler is configured as flow aware the ORDERED and
+	 * ATOMIC schedule queue synchronization by the scheduler is performed
+	 * per flow within each queue.
+	 */
+	odp_support_t flow_aware;
+
+	/* Number of parallel flows per queue to be supported
+	 * Only valid when scheduler is configured as flow_aware
+	 *
+	 * Flows are lightweight entity and packets can be assigned to specific
+	 * flow by the application using odp_packet_flow_hash_set() before
+	 * enqueuing the packet into the scheduler.
+	 * This value is valid only when scheduler is configured in flow aware
+	 * mode.
+	 * Depening on the implementation this number might be rounded-off to
+	 * nearest supported value (e.g power of 2)
+	 * This number should be less than maximum flow supported by the
+	 * implementation.
+	 * @see odp_schedule_capability_t
+	 */
+	uint32_t flow_count;
```

Comment:
It's not clear if count is always rounded down (could it be rounded up to nearest power of two). When rounded, application needs to know the actual value used, so that it's clear which flow ID values are possible / actually affect scheduling.

Application usually wants to spread flows evenly to flow IDs. For example, it makes a difference if flow ID values 0...6 or 0..7 are actually used within scheduler. If application would use value 7, but scheduler would round down 7 to 0 (without application knowing), there would be double the load on flow ID 0 vs other IDs (1..6). When application knows that 7 is not used, it could (potentially) spread those flows evenly to over all flow IDs (0...6).


It could be actually useful to limit this to powers of two

Reply:
@psavol comment still seems valid here.

Reply:
The flow_count is always rounded up to the nearest power of 2.
I agree we need a mechanism to indicate the value selected by the implementation.
Will update in the next version